test/TEST-02-CRYPTSETUP/test.sh

```diff
@@ -1,38 +1,31 @@
 #!/bin/bash
 # -*- mode: shell-script; indent-tabs-mode: nil; sh-basic-offset: 4; -*-
 # ex: ts=8 sw=4 sts=4 et filetype=sh
+set -e
 TEST_DESCRIPTION="cryptsetup systemd setup"
+TEST_NO_NSPAWN=1
 
 . $TEST_BASE_DIR/test-functions
 
 check_result_qemu() {
     ret=1
     mkdir -p $TESTDIR/root
     mount ${LOOPDEV}p1 $TESTDIR/root
-    [[ -e $TESTDIR/root/testok ]] && ret=0
-    [[ -f $TESTDIR/root/failed ]] && cp -a $TESTDIR/root/failed $TESTDIR
+    [ ! -e $TESTDIR/root/testok ] || ret=0
```

Comment:
This doesn't seem to be universally true - with just enabling `set -e` things fail all over the place because of these `cmd1 && cmd2` constructs. It's much simpler and more consistent to avoid this construct altogether really. I can convert them to `if; then`, which is longer, but easier to read and understand.